### DISCUSSION TITLE
Arakawa C grid

### DISCUSSION THREAD

**Discussion by HatYaz** [OpenDrift/opendrift#1315] [Q&A]
Can the OpenDrift platform accept and process data from a numerical model that uses the Arakawa C-grid grid type?

**Comment by knutfrode** [comment]
OpenDrift can work with any grids, provided a suitable reader.

**Reply by paulabirocchi** [reply]
@knutfrode Does the reader_ROMS_native.py script transform the u and v grid points from the Arakawa-C grid (ROMS) to the rho points and then these transformed values are used in Opendrift for the lagrangean dispersion?

**Comment by knutfrode** [comment]
Hi,

The native ROMS reader is available here:
https://github.com/OpenDrift/opendrift/blob/master/opendrift/readers/reader_ROMS_native.py

As you can see from the code/comments, it does not perform any destaggering, so it crudely assumes that the u- and v- (as well as S and T) variables are located at the same positions, provided by the lon- and lat-arrays at rho-points.

But as you can also see, it did include de-staggering at some earlier stage, but I do not remember exactly why this was disabled.

Outside of this reader (generally for all readers), the fields (e.g. current components) are extrapolated towards coast if there should be any gaps, to prevent particles getting stuck with zero currents.
